### PR TITLE
incr.comp.: Don't use Ident in DefPath because that's unstable across compilation sessions

### PR DESCRIPTION
**Pull request by michaelwoerister**
Fixes https://github.com/rust-lang/rust/issues/42550.

cc @jseyfried @nikomatsakis 

r? @eddyb

### CODE REVIEW

File: src/librustc/hir/map/definitions.rs [hunk 1]
```diff
@@ -629,24 +661,20 @@ impl DefPathData {
         }
     }
 
-    pub fn get_opt_name(&self) -> Option<ast::Name> {
-        self.get_opt_ident().map(|ident| ident.name)
-    }
-
     pub fn as_interned_str(&self) -> InternedString {
```
Comment (eddyb) [inline src/librustc/hir/map/definitions.rs]: It'd be nice not to have the duplication between this and `get_opt_name`.

Comment (michaelwoerister) [inline src/librustc/hir/map/definitions.rs]: Would be nice, but they are not quite the same...

**Comment by Mark-Simulacrum** [comment]
This seems like we'd want perf stats for it as well.

@bors p=100

**Pull request merged**